### ISSUE TITLE
MenuItem "Breitenänderung" not working

### ISSUE THREAD

**Issue by rjahn** [sibvisions/reactUI#533]
User: features
Screen: Contacts (DB)

Press button:
<img width="340" alt="image" src="https://github.com/user-attachments/assets/9aa7ace1-7032-47c7-867d-fbb82aab5b6f" />

and item: Breitenänderung.

The column width of Firstname should be changed, because of following code:

```java
	    if (currentWidth != 300)
	    {	        
	        rdbContacts.getRowDefinition().getColumnDefinition("FIRSTNAME").setWidth(300);
	    }
	    else
	    {
	        rdbContacts.getRowDefinition().getColumnDefinition("FIRSTNAME").setWidth(50);
	    }
	    rdbContacts.reload();
```

but nothing happens.

**Comment by gimmixAT** [comment]
The width in the changeColumns triggers an update now. The width seems to be applied wrong though which I'll look into next.

**Comment by gimmixAT** [comment]
the first name columns should now be correctly sized

**Comment by rjahn** [comment]
Works not always:

Open the screen -> Breitenänderung -> OK
(no row is selected in this case)

Drag the column width and do Breitenänderung again -> Not OK

The width request is wrong and doesn't send "width" property to server!

And another problem: If a row is selected and you try to drag the column width -> click event is sent... not right in this case.

**Comment by rjahn** [comment]
If you press the Breitenänderung multiple times it also doesn't work -> should toggle between 300px and 50px

**Comment by gimmixAT** [comment]
toggling back and forth should work now

**Comment by rjahn** [comment]
Not here.
If you open the screen and press the Breitenänderung multiple times, the width toggles, but not between 300px and 50px. It's 395px and 333px.

And if you drag the column, the Breitenänderung does nothing.

**Comment by gimmixAT** [comment]
Regarding the dragging that is because the widths are managed by primereact by that point so our override wont have any effect anymore. The frontend does send a width request to the server but withou the new width and even with the width in the request there is no response from the server that might update the columnwidths.
That only leaves me with the option of resetting the prime column widths every time there is a meta update from the server with column widths, which might cause some unexpected column width changes.

**Comment by rjahn** [comment]
The width request ist just an information for the server. It doesn't send anything back, because the client sets the width and sends the info to the server. It's not necessary to send anything back. Not sendind a width to the server makes the request useless.

**Comment by rjahn** [comment]
You're right. Only happens if you re-open the same screen again by clicking on the menu item.

This is indeed strange.

**Comment by gimmixAT** [comment]
The width should be more exact now because I take the padding of the theme into account. Also the Custom resizing is reset if a column preset changes so that it gets applied properly.

**Comment by rjahn** [comment]
Two points:

1) width request doesn't contain the width -> should
2) The width of the column toggles with "Breitenänderung", but width is different.

The server sends 300px for the width of the column but the client doesn't set 300. In my case, the whole column (th tag) is 325px or 380px. And after first toggle it's 312px and 380px.

It should be 300px or 50px... (independent of point 1)

**Comment by gimmixAT** [comment]
@rjahn could you please try out the `cell-innerwidth` branch. I think the main problem right now is that we set a width but if the content is larger than that the column doesn't shrink. This should be solved in this branch but makes some bigger changes to the table cell markup & styling.

**Comment by mhandsteiner** [comment]
When dragging the width of a column, the width is set with Style:

[data-pc-name="datatable"][pr_id_22] > [data-pc-section="wrapper"] > [data-pc-section="table"] > [data-pc-section="thead"] > tr > th:nth-child(3), [data-pc-name="datatable"][pr_id_22] > [data-pc-section="wrapper"] > [data-pc-section="table"] > [data-pc-section="tbody"] > tr > td:nth-child(3), [data-pc-name="datatable"][pr_id_22] > [data-pc-section="wrapper"] > [data-pc-section="table"] > [data-pc-section="tfoot"] > tr > td:nth-child(3) {
    width: 145px !important;
    max-width: 145px !important;
}

So this could be simply used for setting a fixed width, in case the server sends a width > 0.

There is not the need of setting a width on every header and cell content.

double click on the width spacer could probably remove this style, to get preferred size back.

**Comment by gimmixAT** [comment]
Those styles are generated internally by the Primereact component so we can't really modify them.

In any case the correct width should be sent now with the request

**Comment by mhandsteiner** [comment]
Maybe it is worth to try to handle the insides of the control in handleColResizeEnd.
There are mayby less problems, then trying to work around by limiting the div tags inside all cells. I didn't try the lazy load, but in this workaround also on fetching further rows, all new cells has to be limited to the max size. And when resizing the column manually, the div tags are still cut off. so again the width of all cells has to changed.

When resizing manually, above styles are written in header, and on table the width is restricted.
<table class="p-datatable-table p-datatable-resizable-table" role="table" data-pc-section="table" **style="min-width: 755px; width: 755px;"**>
Removing the width restriction on table, and all not fixed column width styles in head, the table automatically behaves as it should.

It is quiet genious of the base javascript control, that nothing is stored internally, every state is properly stored in dom tree.

So the complete behaviour can be changed from an extended component like ours.

The only thing, we have to solve, is to find out, which column widths are fixed, due to fixed width sent from server, and which columns are fixed due to manual drag in gui by the user. Maybe we have to store a fixed column list internally.

Also, if i would know, how to get the double click on column resizer, i could remove this column from fixed column list, and remove the style, so it would be auto resize again.

**Comment by gimmixAT** [comment]
I fixed some issues in the branch and it should behave better now.

I'll look int oyour suggestions but as I mentioned, the table is very tightly managed by the Primereact UI library and getting in there so far has always been quite hacky.
